test/images/agnhost/webhook/main.go

```diff
@@ -56,6 +57,8 @@ func init() {
 		"File containing the default x509 private key matching --tls-cert-file.")
 	CmdWebhook.Flags().IntVar(&port, "port", 443,
 		"Secure port that the webhook listens on")
+	CmdWebhook.Flags().StringVar(&sidecarImage, "sidecar-image", "k8s.gcr.io/pause:3.1",
```

Comment:
I think we should always require this be passed if /mutating-pods-sidecar is used... we want to avoid e2e tests getting set up not knowing this is required and sometimes not permitted to be k8s.gcr.io

Reply:
The problem is that we read params first and than maybe `/mutating-pods-sidecar` is used so  it's impassible to make `sidecar-image` required under this condition.
I added `klog.Fatal`, I hope it's ok.

Reply:
I would require the command line argument to be passed regardless of whether an endpoint is to be used, so that tests are required to setup properly

Reply:
The question is if I should care about existing e2e tests? If I added a requirement for a parameter, it will make all existing e2e test failing.

Reply:
I think all the tests will take care of their won prereqs, just make sure you override this image explicitly using GetPauseImageName (https://github.com/kubernetes/kubernetes/blob/master/test/utils/image/manifest.go#L278) while starting the webhook server so that anyone can override this image(e.g: windows tests) runtime

Reply:
Based on https://github.com/kubernetes/kubernetes/pull/87385#discussion_r369855032
I assume using GetPauseImageName is not an option.

Reply:
> just make sure you override this image explicitly using GetPauseImageName

I do not think agnhost should set a default here, and should require the sidecar image name be passed. Otherwise, e2e tests will be written not realizing that they have embedded a hard-coded reference to a public registry.

If you make this part of the existing webhook agnhost command, then yes, it means all invokers will need to pass this sidecar flag.

Reply:
yes, thats true. We need to remove that default assignment and pass the image name explicitly.

Reply:
So currently there is no default assignment. If no image is passed, patch will not be applied.

Reply:
> If no image is passed, patch will not be applied.

that seems sensible